app/scripts/session/sessionUtil.js

```diff
@@ -50,17 +50,10 @@
       if (name) {
         _virtualCohort.name = name;
       } else {
-        _virtualCohort.name = "Custom Cohort (" + new Date().toISOString().replace(/T/, ' ') + ")";
-      }
-      if (description) {
-        _virtualCohort.description = description;
-        def.resolve(_virtualCohort);
-      } else {
-        $.when(generateCohortDescription_(cases)).done(function(_desp) {
-          _virtualCohort.description = _desp;
-          def.resolve(_virtualCohort);
-        });
+        _virtualCohort.name = "Selected " + (cases.length > 1 ? "Studies" : "Study");
       }
+      _virtualCohort.description = description || '';
+      def.resolve(_virtualCohort);
```

Comment:
Deferring would no longer be required here. May be we need to clean this in next release

Reply:
It would tricky/unworthy to do to clean up all the code since the return type changed. Will clean up in next release.